packages/commonwealth/client/scripts/views/pages/CommunityManagement/Contests/ContestsList/ContestCard/ContestCard.tsx

```diff
@@ -42,15 +47,34 @@ const ContestCard = ({
   finishDate,
   topics,
   score,
+  decimals,
+  ticker,
   isAdmin,
-  isActive,
+  isCancelled,
   onFund,
 }: ContestCardProps) => {
+  const forceRerender = useForceRerender();
+
   const navigate = useCommonNavigate();
   const { activeAccount: hasJoinedCommunity } = useUserActiveAccount();
 
   const { mutateAsync: cancelContest } = useCancelContestMutation();
 
+  const hasEnded = moment(finishDate) < moment();
+  const isActive = isCancelled ? false : !hasEnded;
+
+  useEffect(() => {
+    if (!isActive) {
+      return;
+    }
+
+    const interval = setInterval(() => {
+      forceRerender();
+    }, 6000); // 6s
+
+    return () => clearInterval(interval);
+  }, [forceRerender, isActive]);
+
```

Comment:
this feels very hacky, why do we want to re-render it like this?

Reply:
because it is interval, it does not need to be hacky ;) there is a countdown happening on the contest card based on the contest end time. If you do not rerender the view, it will display active even though it will be finished.

Reply:
that makes sense. Maybe we could make a hook out of it like `useRerender({ interval: x, condition: y})`. 

Please resolve if this would be over abstraction.

Reply:
created hook, thnkas